test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VisualStudioCredentialProviderTests.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell.Interop;
+using Moq;
+using NuGet.Configuration;
+using NuGet.Credentials;
+using NuGet.VisualStudio;
+using Test.Utility.Threading;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    [Collection(DispatcherThreadCollection.CollectionName)]
+    public class VisualStudioCredentialProviderTests
+    {
+        private static readonly Uri _uri = new Uri("http://unit.test");
+
+        private readonly DispatcherThreadFixture _fixture;
+
+        public VisualStudioCredentialProviderTests(DispatcherThreadFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public void Constructor_ThrowsForNullWebProxyService()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => new VisualStudioCredentialProvider(null));
+
+            Assert.Equal("webProxyService", exception.ParamName);
+        }
+
+        [Fact]
+        public void Id_IsRandomPerInstance()
+        {
+            const int trials = 3;
+            var hashset = new HashSet<string>();
+
+            for (var i = 0; i < trials; ++i)
+            {
+                hashset.Add(new VisualStudioCredentialProvider(Mock.Of<IVsWebProxy>()).Id);
+            }
+
+            Assert.Equal(trials, hashset.Count);
+        }
+
+        [Fact]
+        public async Task GetAsync_ThrowsForNullUri()
+        {
+            var provider = new VisualStudioCredentialProvider(Mock.Of<IVsWebProxy>());
+
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(
+                () => provider.GetAsync(
+                    uri: null,
+                    proxy: Mock.Of<IWebProxy>(),
+                    type: CredentialRequestType.Unauthorized,
+                    message: "a",
+                    isRetry: false,
+                    nonInteractive: true,
+                    cancellationToken: CancellationToken.None));
+
+            Assert.Equal("uri", exception.ParamName);
+        }
+
+        [Fact]
+        public async Task GetAsync_ThrowsIfCancelled()
```

Comment:
To be consistent with the rest test name should follow "Given_When_Then" convention, which in this case would come down to `GetAsync_IfCancelled_Throws`.